### DISCUSSION TITLE
No output on Pimoroni Pico Display Pack (only with certain script)

### DISCUSSION THREAD

**Discussion by Panda2340** [community/community#128291] [Programming Help]
Hi,
I am coding a program for the Raspberry Pi Pico (with a Pimoroni Pico Display Pack), and when I try to run the script, there is no output on the screen. The IDE doesn't give me a syntax error, or an indentation error, but it still outputs on the display pack with no issues when using a different script file.
Here is my script:
```
import time
from pimoroni import Button
from picographics import PicoGraphics, DISPLAY_PICO_DISPLAY, PEN_P4

display = PicoGraphics(display=DISPLAY_PICO_DISPLAY, pen_type=PEN_P4, rotate=0)

display.set_backlight(0.5)
display.set_font("bitmap8")

button_a = Button(12)
button_b = Button(13)
button_x = Button(14)
button_y = Button(15)

WHITE = display.create_pen(255, 255, 255)
BLACK = display.create_pen(0, 0, 0)
CYAN = display.create_pen(0, 255, 255)
MAGENTA = display.create_pen(255, 0, 255)
YELLOW = display.create_pen(255, 255, 0)
GREEN = display.create_pen(0, 255, 0)


def clear():
    display.set_pen(BLACK)
    display.clear()
    display.update()


clear()

display.set_pen(WHITE)
display.text("text", 10, 10, 240, 4)



```
I noticed this right when I ran it after writing the line that makes it output the text:
`display.text("text", 10, 10, 240, 4)`

Please help if you can :)

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by rohispace90** [comment]
Hello dear friend;

It seems like the issue might be related to how the display update is handled after drawing on the screen. In your script, you are using the `picographics` library to control the Pimoroni Pico Display Pack on the Raspberry Pi Pico. Here are a few things to check and adjust:

1. **Display Update:**
   After drawing on the screen using `display.text`, you need to explicitly call `display.update()` to update the display with the changes. This is essential because changes made to the display are not automatically rendered until you explicitly update it.

   Modify your script to include `display.update()` after drawing the text:
   ```python
   display.set_pen(WHITE)
   display.text("text", 10, 10, 240, 4)
   display.update()
   ```

2. **Ensure Display Initialization:**
   Make sure that the display is properly initialized before attempting to draw on it. Based on your script, you are creating an instance of `PicoGraphics`. Ensure that this initialization (`PicoGraphics` instance creation) is correctly executed and that there are no errors in this initialization process.

3. **Check for Errors:**
   Even though you mentioned there are no syntax or indentation errors reported by the IDE, ensure that there are no runtime errors occurring that might prevent the display from updating properly. Check the console output for any runtime errors that might give you clues about what's going wrong.

4. **Library Compatibility:**
   Verify that the version of the `picographics` library and any other related libraries (`pimoroni`, `Button` module) are compatible with your Raspberry Pi Pico and the Pico Display Pack. Sometimes, library versions or compatibility issues can cause unexpected behavior.

5. **Power and Connections:**
   Ensure that your Raspberry Pi Pico is adequately powered and that the Pico Display Pack is connected properly to the GPIO pins. Inadequate power supply or loose connections can sometimes cause peripherals not to function correctly.

Here's how you can adjust your script to include the `display.update()` call after drawing the text:

```
import time
from pimoroni import Button
from picographics import PicoGraphics, DISPLAY_PICO_DISPLAY, PEN_P4

display = PicoGraphics(display=DISPLAY_PICO_DISPLAY, pen_type=PEN_P4, rotate=0)

display.set_backlight(0.5)
display.set_font("bitmap8")

button_a = Button(12)
button_b = Button(13)
button_x = Button(14)
button_y = Button(15)

WHITE = display.create_pen(255, 255, 255)
BLACK = display.create_pen(0, 0, 0)
CYAN = display.create_pen(0, 255, 255)
MAGENTA = display.create_pen(255, 0, 255)
YELLOW = display.create_pen(255, 255, 0)
GREEN = display.create_pen(0, 255, 0)

def clear():
    display.set_pen(BLACK)
    display.clear()
    display.update()

clear()

display.set_pen(WHITE)
display.text("text", 10, 10, 240, 4)
display.update()  # Ensure the display updates after drawing

# Add any additional logic or functionality here

```

By ensuring `display.update()` is called after making changes to the display content, you should see the text "text" appear on your Pico Display Pack screen. If you continue to experience issues, double-check the initialization steps and review any documentation or examples provided by Pimoroni for using the Pico Display Pack with the Raspberry Pi Pico.

Be lucky.

**Reply by Panda2340** [reply]
Thank you, I will try this out, and let you know if this worked.
